### PR TITLE
ART-8041 Use git ls-remote instead of GitHub API

### PR DESCRIPTION
**Pull request by locriandev**
Changes in this PR:
-  `_do_shas_match()` updated to use `git ls-remote` to compare commit SHAs on private/public upstrams without calling GitHub API
- if `ocp4-scan` is built with `DRY_RUN` set to True, pass the `--dry-run` option to `doozer config:scan-sources`; this will not perform the rebase, only print what it would have done
- re-enable the currently disabled rebase procedure, by adding the `--rebase-priv` flag to doozer

Test run: https://saml.buildvm.hosts.prod.psi.bos.redhat.com:8888/job/aos-cd-builds/job/build%252Focp4_scan/553872/console

### CODE REVIEW

**Review by joepvd — APPROVED** [review]
/hold Not sure if special care needs to be take for deploy.

**Comment by locriandev** [comment]
No care is required, can merge right away (with proper supervision which I'll provide)

**Pull request merged**
